### PR TITLE
Updating modulefiles to work with latest spack-stack

### PR DESCRIPTION
**Pull request by mkavulich**
The modulefiles have not been updated in a while and they do not work on some platforms. This PR updates the modulefiles for the latest spack-stack installations (1.6.0) on Derecho, Hera, Jet, and Orion; all the existing. It removes the Jet and Orion GNU modulefiles (I don't think they ever actually worked, these are not officially supported installations). Also, just a few fixes and clarifications for the users guide about spack-stack.

Successfully compiled on Derecho (intel and GNU), Hera (intel and GNU), Jet (Intel) and Orion (Intel). Many regression tests are failing at runtime (35/50), but this is a known problem that will be fixed in other PRs.

Finally, a fix to the supported suites list to remove SCM_RAP (which is no longer supported).

### CODE REVIEW

**Comment by grantfirl** [comment]
Do the modulefile changes correspond to the latest in https://github.com/ufs-community/ufs-weather-model/tree/develop/modulefiles?

**Comment by mkavulich** [comment]
@grantfirl Yes, these changes should correspond to the latest weather model modules (spack-stack 1.6.0)

**Review by dustinswales — COMMENTED** [review]
@mkavulich These changes look good. Just a small comment on a documentation change that I think should be reverted.

File: scm/doc/TechGuide/chap_quick.rst [hunk 1]
```diff
@@ -491,10 +501,11 @@ To see the full list of available options, use the ``--help`` flag:
 
 The run script’s full set of options are described below, where optional abbreviations are included in brackets.
 If using the main branch, you should run the above command to ensure you have the most up-to-date list of options.
+There are no required arguments, but at least one of ``--case`` or ``--file`` must be specified.
```
[the remark this answers is not in the document]
Comment (mkavulich) [reply scm/doc/TechGuide/chap_quick.rst:504]: I don't quite understand why this is confusing, is it the use of the term "required argument"? The script right now fails in a non-intuitive way if users don't specify one of these arguments, so I wanted to add a note about that specific requirement.

Maybe a better way forward is to just fix the script so it gives a clear error message when it doesn't get an argument for ``--file`` or ``--case``, does that sound better?

[the remark this answers is not in the document]
Comment (dustinswales) [reply scm/doc/TechGuide/chap_quick.rst:504]: It's confusing, at least to me, that there are (a) no required arguments, but (b) --case or --file must be specified as an argument.
But yeah, I agree that the better approach is to modify run_scm.py to report error if (not args.case) and (not args.file), and reflect that here in the docs (plus the bit about the default suite).

[the remark this answers is not in the document]
Comment (mkavulich) [reply scm/doc/TechGuide/chap_quick.rst]: Thanks for the feedback; I pushed an update, let me know what you think.

**Review by grantfirl — APPROVED** [review]

**Comment by grantfirl** [comment]
@mkavulich Can you pull in the latest main before we merge this, please?

**Pull request merged**